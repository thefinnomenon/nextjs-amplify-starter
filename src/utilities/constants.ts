export const SITE_URL = 'https://example.com';
export const DEFAULT_OG_IMAGE = 'images/defaultOgImage.png';
export const AUTHOR_NAME = 'Chris Finn';
export const TWITTER_USERNAME = 'thefinnomenon';
