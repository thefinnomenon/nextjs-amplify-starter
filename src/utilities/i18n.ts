/* TODO:
 * I want to replace the static declaration of the locale object by this method that
 * generates it by scanning the locales directory. Unfortunately, `fs` isn't available
 * from `_app.tsx` unless I use getStaticProps which will mess up some optimizations.
 */
import fs from 'fs';
import path, { join } from 'path';

const LOCALES_DIR = join(process.cwd(), 'locales');

// Return an array of the locales and their respective paths - [{ locale: <path> }]
export function getLocalesList(): { [key: string]: string }[] {
  let paths;
  paths = findInDir(LOCALES_DIR, /.json$/);
  console.log(paths);
  return [{ en: '../../locales/en.json' }];
}

// Recurse through directory and return files that match filter
function findInDir(dir: string, filter: RegExp, fileList: string[] = []) {
  const files = fs.readdirSync(dir);

  files.forEach((file) => {
    const filePath = path.join(dir, file);
    const fileStat = fs.lstatSync(filePath);

    if (fileStat.isDirectory()) {
      findInDir(filePath, filter, fileList);
    } else if (filter.test(filePath)) {
      fileList.push(filePath);
    }
  });

  return fileList;
}
