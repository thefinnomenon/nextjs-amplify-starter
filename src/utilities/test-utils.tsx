import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import GlobalStyles from '@/styles/themeConfig';
import { ThemeProvider } from 'next-themes';
import { IntlProvider } from 'react-intl';

const locale = 'en';

// Load locale files
const locales = {
  en: require('@/locales/en.json'),
  pt: require('@/locales/pt.json'),
};

const messages = locales[locale];

const AllTheProviders: FC = ({ children }) => {
  return (
    <ThemeProvider>
      <GlobalStyles />
      <IntlProvider messages={messages} locale={locale}>
        {children}
      </IntlProvider>
    </ThemeProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
