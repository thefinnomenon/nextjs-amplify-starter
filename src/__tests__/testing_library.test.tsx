/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@/utilities/test-utils';
import Index from '../pages/index';

import MatchMediaMock from 'jest-matchmedia-mock';
let matchMedia;

describe('App', () => {
  beforeAll(() => {
    matchMedia = new MatchMediaMock();
  });

  afterEach(() => {
    matchMedia.clear();
  });

  it('renders homepage unchanged', () => {
    const { container } = render(<Index />);
    expect(container).toMatchSnapshot();
  });

  it('renders navigation', () => {
    const { getByRole } = render(<Index />);

    // eslint-disable-next-line testing-library/prefer-screen-queries
    const nav = getByRole('navigation');

    expect(nav).toBeInTheDocument();
  });

  it('renders one link', () => {
    const { getByRole } = render(<Index />);

    // eslint-disable-next-line testing-library/prefer-screen-queries
    const nav = getByRole('link', { name: 'One' });

    expect(nav).toBeInTheDocument();
  });
});
