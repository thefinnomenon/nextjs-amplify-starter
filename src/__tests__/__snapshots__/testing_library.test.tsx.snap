// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders homepage unchanged 1`] = `
<div>
  <button
    class="ScrollToTop__Wrapper-sc-wqa3ky-0 giQJY"
  >
    <div
      class="ScrollToTop__DefaultScrollToTop-sc-wqa3ky-1 fJrTWa"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-up fa-w-14 "
        data-icon="arrow-up"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z"
          fill="currentColor"
        />
      </svg>
    </div>
  </button>
  <div
    class="SingleColumnLayout__Wrapper-sc-1wn70mv-0 bOndOx"
  >
    <div
      class="Header__Wrapper-sc-18nvbin-0 cVHQkk"
    >
      <a
        class="Header__LogoLink-sc-18nvbin-1 gifGvv"
        href="/"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjQwIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            alt="Site logo"
            class="Header__Logo-sc-18nvbin-2"
            data-nimg="true"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
      </a>
      <div
        class="Header__Controls-sc-18nvbin-6 bxHiaj"
      >
        <svg
          color="black"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="cursor: pointer; transform: rotate(90deg);"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="circle-mask-1"
          >
            <rect
              fill="white"
              height="100%"
              width="100%"
              x="0"
              y="0"
            />
            <circle
              fill="black"
              r="9"
            />
          </mask>
          <circle
            cx="12"
            cy="12"
            fill="black"
            mask="url(#circle-mask-1)"
          />
          <g
            stroke="currentColor"
            style="opacity: 1;"
          >
            <line
              x1="12"
              x2="12"
              y1="1"
              y2="3"
            />
            <line
              x1="12"
              x2="12"
              y1="21"
              y2="23"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="4.22"
              y2="5.64"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="18.36"
              y2="19.78"
            />
            <line
              x1="1"
              x2="3"
              y1="12"
              y2="12"
            />
            <line
              x1="21"
              x2="23"
              y1="12"
              y2="12"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="19.78"
              y2="18.36"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="5.64"
              y2="4.22"
            />
          </g>
        </svg>
        <select
          class="LanguageControl__Select-sc-10zgvl6-0 cnulBu"
          height="20"
        >
          <option
            selected=""
            value="en"
          >
            EN
          </option>
        </select>
      </div>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-hamburger fa-w-16 fa-lg Header__HamburgerMenu-sc-18nvbin-3 ZwHUJ"
        data-icon="hamburger"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 256H48a48 48 0 0 0 0 96h416a48 48 0 0 0 0-96zm16 128H32a16 16 0 0 0-16 16v16a64 64 0 0 0 64 64h352a64 64 0 0 0 64-64v-16a16 16 0 0 0-16-16zM58.64 224h394.72c34.57 0 54.62-43.9 34.82-75.88C448 83.2 359.55 32.1 256 32c-103.54.1-192 51.2-232.18 116.11C4 180.09 24.07 224 58.64 224zM384 112a16 16 0 1 1-16 16 16 16 0 0 1 16-16zM256 80a16 16 0 1 1-16 16 16 16 0 0 1 16-16zm-128 32a16 16 0 1 1-16 16 16 16 0 0 1 16-16z"
          fill="currentColor"
        />
      </svg>
      <nav
        class="Header__NavItems-sc-18nvbin-4 kpKKUt"
      >
        <a
          class="Header__NavItem-sc-18nvbin-5 kRHTgL"
          href="/one"
        >
          One
        </a>
        <a
          class="Header__NavItem-sc-18nvbin-5 kRHTgL"
          href="/two"
        >
          Two
        </a>
        <a
          class="Header__NavItem-sc-18nvbin-5 kRHTgL"
          href="/three"
        >
          Three
        </a>
      </nav>
    </div>
    <main
      class="SingleColumnLayout__Content-sc-1wn70mv-1 cGUkOK"
    >
      <h1>
        Hi!
      </h1>
    </main>
  </div>
</div>
`;
